examples/gpu-printing/main.cpp

```diff
@@ -201,21 +198,12 @@ Result execute()
 
 };
 
-// This "inner" main function is used by the platform abstraction
-// layer to deal with differences in how an entry point needs
-// to be defined for different platforms.
-//
-void innerMain(ApplicationContext* context)
+int main()
```

Comment:
Design nitpick: the choice to wrap the `main()` function even for console applications was a conscious choice.

Most importantly, on Windows we are likely to need/want a `wmain()` entry point for Unicode support, and that entry point will need to convert all the command-line arguments from their wide character encoding to UTF-8. The inner main should not have to think about those details, and we want our command-line apps to be Unicode-compliant by default.

There's also some details here on many Linux platforms where we likely want to set the C locale to a region-independent but Unicode-compatible locale like `"C.UTF08"`.

Having a distinct "context" type here that gets passed down into the code also avoids the issue of having an "ambient authority" - it makes it so that application code (including things like plugins) need to have the relevant platform-specific objects passed in directly in order to interact with them.

Eventually we may need/want things like the application context to be a COM interface (like what we do for the compiler and `gfx` APIs), so that we can easily expose varying per-target facilities (including ones that cannot just be detected at compile time).